Source/Classes/PINRequestRetryStrategy.m

```diff
@@ -0,0 +1,69 @@
+//
+//  PINRequestRetryStrategy.m
+//  Pods
+//
+//  Created by Hovhannes Safaryan on 9/24/16.
+//
+//
+
+#import "PINRequestRetryStrategy.h"
+#import "PINURLSessionManager.h"
+#import "PINRemoteImageManager.h"
+
+@interface PINRequestExponentialRetryStrategy ()
+
+@property (nonatomic, assign) int retryMaxCount;
+@property (nonatomic, assign) int retryCount;
+@property (nonatomic, assign) int delayBase;
+
+@end
+
+@implementation PINRequestExponentialRetryStrategy
+
+- (instancetype)initWithRetryMaxCount:(int)retryMaxCount delayBase:(int)delayBase
+{
+    if (self = [super init]) {
+        _retryCount = 0;
+        _retryMaxCount = retryMaxCount;
+        _delayBase = delayBase;
+    }
+    return self;
+}
+
+- (int)numberOfRetries
+{
+    return self.retryCount;
+}
+
+- (BOOL)shouldRetryWithError:(NSError *)error
+{
+    if (error == nil || ![[self class] retriableError:error] ||
+        self.retryCount > self.retryMaxCount) {
```

Comment:
retryMaxCount is the count to be retried, so I think it's right now, if we change this to `>=`, this will be try max count(not retry), or something like that :) Let me know should be change this to `>=`.